src/plugins/github/heuristics/mentionsAuthorReference.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+
+import * as MapUtil from "../../../util/map";
+import * as RV from "../relationalView";
+import * as N from "../nodes";
+
+// MentionsAuthorReferences are inferred edges between posts.  If a post
+// contains an @-mention of a user, and that user has authored any posts
+// in the same thread, then we create MentionsAuthorReferences from the
+// mentioning post to any posts authored by the mentioned author.  A
+// concrete example: Suppose I write a post, and I @-mention you. Then
+// my post will have MentionsAuthorReferences pointing to any posts that
+// you write. The inference is that because I mentioned you by name, my
+// post probably values your posts (either because I am referring to
+// content you wrote in a previous post, or because I appreciate that
+// you showed up to engage with me after being mentioned)
+//
+// See #804 for context:
+// https://github.com/sourcecred/sourcecred/issues/804
+
+export type MentionsAuthorReference = {|
+  // The post that generated an mentions author reference (via @-mention)
+  +src: N.TextContentAddress,
+  // The post authored by the person mentioned in the src
+  +dst: N.TextContentAddress,
+  // The person that was mentioned in src.  Storing `who` is necessary
+  // because in the case of paired authorship, there might be multiple
+  // pointers from the same src to the same dst
+  +who: N.UserlikeAddress,
+|};
+
+export function* findMentionsAuthorReferences(
```

Comment:
I like that this and its return type are the only two exports. Good encapsulation.